src/topp/CometAdapter.cpp

```diff
@@ -141,43 +141,26 @@ class TOPPCometAdapter :
     setValidStrings_(Constants::UserParam::ISOTOPE_ERROR, ListUtils::create<String>("off,0/1,0/1/2,0/1/2/3,-8/-4/0/4/8"));
 
-    //Search Enzyme
-    vector<String> all_enzymes;
-    ProteaseDB::getInstance()->getAllCometNames(all_enzymes);
-    registerStringOption_("enzyme", "<cleavage site>", "Trypsin", "The enzyme used for peptide digestion.", false, false);
-    setValidStrings_("enzyme", all_enzymes);
-    registerStringOption_("second_enzyme", "<cleavage site>", "", "The enzyme used for peptide digestion.", false, true);
-    setValidStrings_("second_enzyme", all_enzymes);
-
-    registerStringOption_("num_enzyme_termini", "<choice>", "fully", "Specify the termini where the cleavage rule has to match", false, false);
-    setValidStrings_("num_enzyme_termini", ListUtils::create<String>("semi,fully,C-term unspecific,N-term unspecific"));
-    registerIntOption_("allowed_missed_cleavages", "<num>", 0, "Number of possible cleavage sites missed by the enzyme. It has no effect if enzyme is unspecific cleavage.", false, false);
-    setMinInt_("allowed_missed_cleavages", 0);
-    setMaxInt_("allowed_missed_cleavages", 5);
-
-    registerIntOption_("min_peptide_length", "<num>", 5, "Minimum peptide length to consider.", false);
-    setMinInt_("min_peptide_length", 5);
-    setMaxInt_("min_peptide_length", 63);
-    registerIntOption_("max_peptide_length", "<num>", 63, "Maximum peptide length to consider.", false);
-    setMinInt_("max_peptide_length", 5);
-    setMaxInt_("max_peptide_length", 63);
-
     //Fragment Ions
-    registerDoubleOption_("fragment_bin_tolerance", "<tolerance>", 0.02, "Bin size (in Da) for matching fragment ions. Ion trap: 1.0005, high res: 0.02. CAUTION: Low tolerances have heavy impact on RAM usage. Consider using use_sparse_matrix and/or spectrum_batch_size.", false, true);
-    setMinFloat_("fragment_bin_tolerance", 0.01);
-    registerDoubleOption_("fragment_bin_offset", "<fraction>", 0.0, "Offset of fragment bins scaled by tolerance. Ion trap: 0.4, high res: 0.0.", false, true);
+    registerDoubleOption_("fragment_mass_tolerance", "<tolerance>", 0.3,
+                          "Fragment mass error (one-sided)."
+                          "CAUTION: Low tolerances have heavy impact on RAM usage (since Comet uses a lot of bins in this case). Consider using use_sparse_matrix and/or spectrum_batch_size.", false);
+    setMinFloat_("fragment_mass_tolerance", 0.01);
+    registerStringOption_("fragment_error_units", "<unit>", "Da", "Fragment monoisotopic mass error units", false);
+    setValidStrings_("fragment_error_units", { "Da" }); // only Da allowed
+    registerDoubleOption_("fragment_bin_offset", "<fraction>", 0.0, "Offset of fragment bins. Recommended by Comet: low-res: 0.4, high-res: 0.0", false);
```

Comment:
this is the offset for high-res but tolerance up there is for low-res ?

Reply:
good catch